screens/ChatScreen.js

```diff
@@ -42,126 +44,249 @@ class ChatScreen extends Component {
+    console.log(requestUri);
+
+    // TODO: add device ID
+    const bodyData = { device: 'test' };
+    const response = await fetch(requestUri, { method: 'post', body: JSON.stringify(bodyData), headers: { 'Content-Type': 'application/json' } });
+    const responseJson = await response.json();
 
-    const channelHandler = new this.sendBirdApp.ChannelHandler();
-    channelHandler.onMessageReceived = this.onReceive;
-    this.sendBirdApp.addChannelHandler('ChatScreen');
+    console.log('Printing server results');
+    console.log(responseJson);
 
-    // Get previous messages
-    const query = this.channel.createPreviousMessageListQuery();
-    const messages = await new Promise(resolve => {
-      query.load(30, false, (err, msgs) => {
-        resolve(msgs);
-      })
-    })
+    const twilioToken = responseJson.chat_token;
+    return twilioToken;
+  }
 
-    console.log("Map state");
+  getChannelName() {
+    const localToken = JSON.parse(this.state.hmToken);
+    const contact = this.props.navigation.state.params.mentee._id;
+    return localToken._id > contact ? `${localToken._id}.${contact}` : `${contact}.${localToken._id}`;
+  }
 
-    // Map them to GiftedChat format
-    this.setState({
-      messages: messages.map(m => {
-        return {
-          _id: m.messageId.toString(),
-          text: m.message,
-          createdAt: m.createdAt,
-          user: {
-            _id: m.sender.userId
-          }
+  async getMessage() {
+    this.channel.then((c) => {
+      c.on('messageAdded', message => this.updateLocalMessageStateSingle(message));
+
+      c.getMessages().then((messages) => {
+        const totalMessages = messages.items.length;
+        const localMessages = [];
+
+        for (let i = 0; i < totalMessages; i += 1) {
+          const message = messages.items[i];
+          const currMessage = {
+            _id: `${message.index}`,
+            text: `${message.body}`,
+            createdAt: `${message.timestamp}`,
+            user: {
+              _id: `${message.author}`,
+            },
+          };
+
+          localMessages.push(currMessage);
         }
-      })
+
+        this.setState(previousState => ({
+          messages: previousState.messages.concat(localMessages),
+        }));
+      });
     });
+  }
 
-    this.setState({ sendBirdApp: sendBirdApp });
+  async updateLocalMessageStateSingle(message) {
+    const messages = {
+      items: [message]
+    };
 
-    console.log("Done mapping state"); */
+    this.updateLocalMessageState(messages);
   }
 
-  async onSend(messages = []) {
-    const promises = messages.map(msg => new Promise((resolve) => {
-      this.channel.sendUserMessage(msg.text, resolve);
-    }));
-    await Promise.all(promises);
+  async updateLocalMessageState(messages) {
+    const totalMessages = messages.items.length;
+    const localMessages = [];
+
+    for (let i = 0; i < totalMessages; i += 1) {
```

Comment:
Use map function same as above. I really try to avoid using for loops in Javascript since it's error prone and almost every problem can be solved with map, reduce, some, every, etc. if you really need a for loop use forEach

Reply:
Fixed